### PR TITLE
Fix a couple Airflow release doc errors

### PR DESCRIPTION
**Pull request by jedcunningham**

### CODE REVIEW

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -583,15 +583,14 @@ Check if sources are the same as in the tag:
...
+pushd apache-airflow-X.Y.Z
 diff -r airflow "${SOURCE_DIR}"
-popd && rm -rf apache-airflow-X.Y.Zrc1
+popd && rm -rf apache-airflow-X.Y.Z
 ```
 
 The output should only miss some files - but they should not show any differences in the files:
 
 ```
-⌂6.50 [jarek:~/asf-dist/dev/airflow/2.6.2rc2/a] * 1 ‡ diff -r airflow ~/code/airflow/
```
Comment (hussein-awala) [inline dev/README_RELEASE_AIRFLOW.md:594]: I think we need to keep just the command, no?

**Review by phanikumv — COMMENTED** [review]
can you please mention the context of why we are doing this(in the description), is it because the last time RC packaging wasn't correct ?

**Review by potiuk — APPROVED** [review]

File: dev/README_RELEASE_AIRFLOW.md [hunk 2]
```diff
@@ -583,15 +583,14 @@ Check if sources are the same as in the tag:
...
-pushd apache-airflow-X.Y.Zrc1
+pushd apache-airflow-X.Y.Z
 diff -r airflow "${SOURCE_DIR}"
-popd && rm -rf apache-airflow-X.Y.Zrc1
+popd && rm -rf apache-airflow-X.Y.Z
 ```
 
 The output should only miss some files - but they should not show any differences in the files:
 
 ```
```
Comment (potiuk) [inline dev/README_RELEASE_AIRFLOW.md:593]:
```suggestion
diff -r airflow "${SOURCE_DIR}"
```

**Comment by potiuk** [comment]
Approved with the command restoring :)

**Comment by potiuk** [comment]
> can you please mention the context of why we are doing this(in the description), is it because the last time RC packaging wasn't correct ?

We usually place little context in case we correct those instructions @phanikumv :). The thing which such instructions (which you learn when you start doing release manager duty) is that pretty much **every-single-time** (TM)  when you follow such instructions you find typos and mistakes, and changed circumstances that made previous instructions **slightly wrong**.

This means that every time the context wil be "Found those small typoes with last iteration of following the process". There are possibly tens or hundreds similar commits in the past :).

This is more-or-less fact of life we have to deal with until you just make it part of a fully automated CI/release process (in which case also things stop working occasionally and you need to fix it occasionally). 

The good thing is that I hope in about 6-8 months the ASF will have a way to automate some of the release process so that we do not have to do it manually on "release-manager exclusively administered machines" (currently this is a requirement by the ASF) and hopefully, we will be able to make it more of trigger in a CI that we will keep on running on a daily basis.

File: dev/README_RELEASE_AIRFLOW.md (hunk 2 above)
Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md:593]: Note @jedcunningham -> no idea how to suggest ``` in the suggestion so just do it manually

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (jedcunningham) [reply dev/README_RELEASE_AIRFLOW.md:594]: That was what I did initially actually, but after thinking about it, I don't think we need it. It already mentions the "output", meaning it's the result of the commands above. This is the same pattern we have for the [shasum check](https://github.com/apache/airflow/blob/main/dev/README_RELEASE_AIRFLOW.md#sha512-sum-check) too.

File: dev/README_RELEASE_AIRFLOW.md (hunk 2 above)
Comment (jedcunningham) [reply dev/README_RELEASE_AIRFLOW.md:593]: See https://github.com/apache/airflow/pull/31951#discussion_r1232405802

**Review by potiuk — APPROVED** [review]

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (jedcunningham) [reply dev/README_RELEASE_AIRFLOW.md:594]: Is this acceptable, or do folks still want the command in here? @potiuk @hussein-awala?

Comment (hussein-awala) [reply dev/README_RELEASE_AIRFLOW.md:594]: It's okay for me, LGTM

**Review by pierrejeambrun — APPROVED** [review]

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (hussein-awala) [reply dev/README_RELEASE_AIRFLOW.md:594]: I just realized that my first review was a comment and not an approve

**Review by hussein-awala — APPROVED** [review]
I just realized that my first review was a comment and not an approve

File: dev/README_RELEASE_AIRFLOW.md (hunk 1 above)
Comment (jedcunningham) [reply dev/README_RELEASE_AIRFLOW.md:594]: Haha no worries, thanks!

**Pull request merged**